File: backend/MonteCarloEarth/Model/Point.cs
using System;
using Microsoft.AspNetCore.Mvc.ModelBinding;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using MonteCarloEarth.Common;

namespace MonteCarloEarth.Model
{
    public class Point : IPoint
    {
        private Point()
        {

        }

        [BsonId]
        [BindNever]
        [BsonElement("id")]
        [BsonRepresentation(BsonType.String)]
        public string Id { get; set; }

        [BsonElement("latitude")]
        public float Latitude { get; set; }

        [BsonElement("longitude")]
        public float Longitude { get; set; }

        [BsonElement("IsWater")]
        public bool IsWater { get; set; }

        public static Point CreatePoint()
        {
            Random r = new Random();
            float latitude = MapToFloat(r.NextDouble(), -90, 90);
            float longitude = MapToFloat(r.NextDouble(), -180, 180);
            return new Point
            {
                Latitude = latitude,
                Longitude = longitude,
                Id = Guid.NewGuid().ToString()
            };
        }

        //maps double in range (0, 1) to float in range (min, max)
        private static float MapToFloat(double value, double min, double max)
        {
            double range = max - min;
            return (float)(value * range + min);
        }
    }
}

File: golang/model/Point.go
package model

import "math/rand"

type Point struct {
	Latitude  float32 `json:"latitude"`
	Longitude float32 `json:"longitude"`
	OnWater bool `json:"isWater"`
}

func NewPoint(latitude, longitude float32, onWater bool) Point {
	return Point {Latitude: latitude, Longitude:longitude, OnWater:onWater}
}

func RandomPoint() Point {
	latitude := mapCoord(rand.Float32(), -90, 90)
	longitude := mapCoord(rand.Float32(), -180, 180)
			//Id = ObjectId.GenerateNewId().ToString()

	return NewPoint(latitude, longitude, false)
}

func mapCoord(coord, min, max float32) float32 {
	span := max - min
	return coord * span + min
}